<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>607</width>
    <height>574</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton {
    background-color: #2d89ef;
    color: white;
    border-radius: 6px;
    padding: 6px 12px;
}
QPushButton:disabled {
    background-color: #aaaaaa;
    color: #dddddd;
}
QPushButton:hover {
    background-color: #218838;
}
QPushButton:pressed {
    background-color: #1e7e34;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>611</width>
      <height>571</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Strona</string>
     </attribute>
     <widget class="QTextBrowser" name="textBrowser_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>601</width>
        <height>521</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;h2 style=&quot; margin-top:16px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:x-large; font-weight:700;&quot;&gt;Neural Network Inference Tool&lt;/span&gt;&lt;/h2&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This program loads a simple feed-forward neural network model from a text file, parses its structure and parameters, and runs inference on the input datadefined in the same file. The outputs of the network are displayed in a clear, tabular format for inspection.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The program expects the neural network and input data to be stored in a plain text file. The file must follow this structure:&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'monospace';&quot;&gt;INPUT 4&lt;br /&gt;LAYER DENSE 3 relu&lt;br /&gt;WEIGHTS:&lt;br /&gt;0.1 0.2 0.3 0.4&lt;br /&gt;0.5 0.6 0.7 0.8&lt;br /&gt;0.9 1.0 1.1 1.2&lt;br /&gt;BIASES:&lt;br /&gt;0.01 0.02 0.03&lt;br /&gt;LAYER DENSE 2 softmax&lt;br /&gt;WEIGHTS:&lt;br /&gt;0.1 0.2 0.3&lt;br /&gt;0.4 0.5 0.6&lt;br /&gt;BIASES:&lt;br /&gt;0.0 0.0&lt;br /&gt;&lt;br /&gt;=== DATA ===&lt;br /&gt;0.5 0.1 0.0 0.9&lt;/span&gt;&lt;/p&gt;
&lt;h3 style=&quot; margin-top:14px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:large; font-weight:700;&quot;&gt;Format description&lt;/span&gt;&lt;/h3&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;
&lt;li style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;INPUT n&lt;/span&gt; - defines the input vector size.  &lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;LAYER DENSE m activation&lt;/span&gt; – adds a fully connected layer with &lt;span style=&quot; font-style:italic;&quot;&gt;m&lt;/span&gt; neurons and the given activation function.&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;WEIGHTS:&lt;/span&gt; - followed by &lt;span style=&quot; font-style:italic;&quot;&gt;m&lt;/span&gt; rows, each with &lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt; numbers (the weight matrix for the layer).  &lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;BIASES:&lt;/span&gt; - one row with &lt;span style=&quot; font-style:italic;&quot;&gt;m&lt;/span&gt; numbers (the bias vector).  &lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;=== DATA ===&lt;/span&gt; - section with input vectors, one per line.&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;All values must be separated by spaces. Lines starting with &lt;span style=&quot; font-family:'Courier New';&quot;&gt;#&lt;/span&gt; are ignored.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Model</string>
     </attribute>
     <widget class="QPushButton" name="loadFile">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>150</y>
        <width>541</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Load model from file...</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="fileNameField">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>541</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="runButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>480</y>
        <width>541</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Run inference</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="modelInfo">
      <property name="geometry">
       <rect>
        <x>33</x>
        <y>189</y>
        <width>541</width>
        <height>281</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="textBrowser">
      <property name="geometry">
       <rect>
        <x>5</x>
        <y>10</y>
        <width>591</width>
        <height>91</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">h3 {
    margin-bottom: 0px;
}

ol, li {
    margin-top: 0px;
}</string>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;h3 style=&quot; margin-top:14px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:large; font-weight:700;&quot;&gt;Instructions&lt;/span&gt;&lt;/h3&gt;
&lt;ol style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Click &lt;span style=&quot; font-weight:700;&quot;&gt;Load file…&lt;/span&gt; and select a file with neuron network model.  &lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;After loading, the &lt;span style=&quot; font-weight:700;&quot;&gt;Run inference&lt;/span&gt; button will be enabled.  &lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Click &lt;span style=&quot; font-weight:700;&quot;&gt;Run inference&lt;/span&gt; to see the results in the &lt;span style=&quot; font-weight:700;&quot;&gt;Output&lt;/span&gt; tab.&lt;/li&gt;&lt;/ol&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Output</string>
     </attribute>
     <widget class="QPlainTextEdit" name="outputView">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>601</width>
        <height>521</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>10</pointsize>
       </font>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
